controller/DonateController.js

```diff
@@ -0,0 +1,57 @@
+const { Donate } = require("../services");
+const catchAsyncError = require("../middleware/catchAsyncError");
+
+/**
+ * @description Donate
+ * @route POST /v1/donate
+ * @access PUBLIC
+ */
+exports.donateNow = catchAsyncError(async (req, res) => {
+  /**
+   * @todo Handle validation for req.body
+   */
+
+  const donation = await Donate.makeDonation(req.body);
+  if (donation.success === true)
+    return res.status(201).json({
+      status: true,
+      message: "Follow this link to complete donation",
+      data: donation.data
+    });
+  else
+    return res.status(500).json({
+      status: false,
+      message: "Unfortunately, something happened"
+    });
+});
+
+/**
+ * @description Flutterwave web hook
+ * @route POST /v1/donate/flw-webhook
+ */
+exports.flw_Webhook = catchAsyncError(async (req, res) => {
+  const secretHash = process.env.FLW_SECRET_HASH;
```

Comment:
Let's move this validation inside of a middleware so that all pre-validation logic sits there.
That's a common pattern we are already using in the codebase.

See this comment https://github.com/edustipend/edustipend-api/pull/90/files#r1582910484

Reply:
Do I include the one for verifying transactions?